src/DotLiquid/FileSystems/LocalFileSystem.cs

```diff
@@ -37,14 +40,13 @@ public string ReadTemplateFile(Context context, string templateName)
 
         public string FullPath(string templatePath)
         {
-            if (templatePath == null || !Regex.IsMatch(templatePath, @"^[^.\/][a-zA-Z0-9_\/]+$"))
+            if (templatePath == null || !Regex.IsMatch(templatePath, @"^(?![\\\/\.])(?:[^<>:;,?""*|\x00-\x1F\/\\]+|[\/\\](?!\.))+(?<!\/)$"))

```

Comment:
code should be refactored to avoid copies.

Reply:
Generally, we have all regex strings declared in Liquid.cs, but probably better to address by declaring the regex somewhere common. The rest of the fullpath implementations differ due to path separators being different.